src/network.c

```diff
@@ -694,40 +694,60 @@ size_t readn_raw(int fd, void *buf, size_t amount) {
   struct timeval sel_tv;
   fd_set rfd;
 
-  FD_ZERO(&rfd);  // initialize with zeroes
+  FD_ZERO(&rfd);
   FD_SET(fd, &rfd);
-  sel_tv.tv_sec = 600;
-  sel_tv.tv_usec = 0;
-
-  /* Using a select() call to timeout if no read occurs after 10 minutes of waiting.
-   * This should fix a bug where the server hangs at it looks like it's in this read
-   * state.  The select() should check to see if there is anything to read on this socket.
-   * if not, and the 3 second timer goes off, exit out and clean up.
-   *
-   * Now that the server does no reading of any sockets (only the child process
-   * reads any sockets), this fix may no longer be necessary, as a long wait
-   * will eventually get interrupted by the child's alarm().
-   */
+
+  // Using a select() call to timeout if no read occurs after 10 seconds of
+  // waiting.  This should fix a bug where the server hangs during read().  The
+  // select() should check to see if there is anything to read on this socket.
+  // If not, and the 10 second timer goes off, exit out and clean up.
+  //
+  // Now that the server does no reading of any sockets (only the child process
+  // reads any sockets), this fix may no longer be necessary, as a long wait
+  // should eventually get interrupted by the child's alarm().
   while (received < amount) {
-    // check if fd+1 socket is ready to be read
+    // The values for the timeout are undefined after select() is called, so we
```

Comment:
My opinion is that this loop could be simpler. The function breakdown could fall along similar lines as writen_any.